### PR TITLE
chore: save build template file after code gen

### PR DESCRIPTION
**Pull request by Mahefaa**

### CODE REVIEW

File: src/main/java/api/jcloudify/app/repository/jpa/EnvDeploymentConfRepository.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package api.jcloudify.app.repository.jpa;
+
+import api.jcloudify.app.repository.model.EnvDeploymentConf;
+import org.springframework.data.jpa.repository.JpaRepository;
+
```
Comment (Onitsiky) [inline src/main/java/api/jcloudify/app/repository/jpa/EnvDeploymentConfRepository.java:5]: @Repository

File: src/main/java/api/jcloudify/app/service/event/PojaConfUploadedService.java [hunk 2]
```diff
@@ -43,13 +50,20 @@
 @AllArgsConstructor
 @Slf4j
 public class PojaConfUploadedService implements Consumer<PojaConfUploaded> {
+  private static final String BUILD_TEMPLATE_FILENAME_YML = "template.yml";
+  private static final String CF_STACKS_CD_COMPUTE_PERMISSION_YML =
```
Comment (Onitsiky) [inline src/main/java/api/jcloudify/app/service/event/PojaConfUploadedService.java:54]: ..._PATH ?

File: src/main/java/api/jcloudify/app/service/event/PojaConfUploadedService.java [hunk 3]
```diff
@@ -43,13 +50,20 @@
...
+      "cf-stacks/cd-compute-permission.yml";
+  private static final String CF_STACKS_EVENT_STACK_YML = "cf-stacks/event-stack.yml";
```
Comment (Onitsiky) [inline src/main/java/api/jcloudify/app/service/event/PojaConfUploadedService.java:56]: idem

File: src/main/java/api/jcloudify/app/service/event/PojaConfUploadedService.java [hunk 4]
```diff
@@ -43,13 +50,20 @@
...
+  private static final String CF_STACKS_STORAGE_BUCKET_STACK_YML =
```
Comment (Onitsiky) [inline src/main/java/api/jcloudify/app/service/event/PojaConfUploadedService.java:57]: idem

File: src/main/resources/db/migration/V42_30__Create_env_deployment_conf_table.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
+create table env_deployment_conf
+(
+    id                                varchar primary key                              default uuid_generate_v4(),
+    compute_permission_stack_file_key varchar,
+    event_stack_file_key              varchar,
+    storage_bucket_stack_file_key     varchar,
+    build_template_file               varchar,
+    env_id                            varchar references "environment" ("id") not null,
```
Comment (Onitsiky) [inline src/main/resources/db/migration/V42_30__Create_env_deployment_conf_table.sql:8]: independently define the foreign key constraint

**Review by Onitsiky — CHANGES_REQUESTED** [review]

File: src/main/java/api/jcloudify/app/repository/jpa/EnvDeploymentConfRepository.java (hunk 1 above)
Comment (Mahefaa) [reply src/main/java/api/jcloudify/app/repository/jpa/EnvDeploymentConfRepository.java:5]: there's no need to add @Repository here

**Review by Onitsiky — APPROVED** [review]

**Pull request merged**